File: Backend/app/models/history.py
from django.db import models

from .auction_user import AuctionUser
from .item import Item


class History(models.Model):
    description = models.TextField(null=True, blank=True)
    auction_user = models.ForeignKey(AuctionUser, on_delete=models.CASCADE, related_name="histories")
    item = models.ForeignKey(Item, on_delete=models.CASCADE, related_name="histories", null=True, blank=True)
    amount = models.PositiveIntegerField(default=0)

File: backend/src/models/history.js
const { DataTypes } = require('sequelize');

const sequelize = require('../../conf/sequelize');
const AuctionUser = require('./auction_user');
const User = require('./user');


const History = sequelize.define('history', {
    id: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true,
    },
    description: {
        type: DataTypes.TEXT,
        allowNull: false,
    },
    amount: {
        type: DataTypes.DOUBLE,
        defaultValue: 0.0,
    },
    type: {
        type: DataTypes.STRING,
        allowNull: false,
    }
},
    {
        tableName: 'history',
    }
);

History.belongsTo(AuctionUser, {foreignKey: "auction_user_id"});
AuctionUser.hasOne(History, {foreignKey: "auction_user_id"})

History.belongsTo(User, { foreignKey: 'user_id' })
User.hasOne(History, { foreignKey: 'user_id' });

module.exports = History;
